hyperstart.go

```diff
@@ -334,14 +373,57 @@ func (h *hyper) startContainer(pod Pod, contConfig ContainerConfig) error {
 		return err
 	}
 
+	err = h.exec(pod, contConfig, contConfig.Cmd)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
 // stopContainer is the agent Container stopping implementation for hyperstart.
 func (h *hyper) stopContainer(pod Pod, container Container) error {
-	payload := []byte(fmt.Sprintf("{\"container\":\"%s\",\"signal\":\"9\"}", container.id))
+	cmd := Cmd{
+		Args:    []string{"/bin/pkill", h.buildPauseBinName(pauseBinName, pod.id, container.id)},
+		Envs:    container.config.Cmd.Envs,
+		WorkDir: "/",
+	}
+
+	err := h.exec(pod, *(container.config), cmd)
+	if err != nil {
+		return err
+	}
+
+	err = h.bindUnmountContainerRootfs(*(container.config))
+	if err != nil {
+		return err
+	}
+
+	restartPod := true
+	for _, c := range pod.containers {
+		state, err := fetchContainerState(&pod, c.ID)
+		if err != nil {
+			return err
+		}
+
+		if c.ID == container.id {
+			continue
+		} else if c.ID != container.id && state.State == stateReady {
+			continue
+		}
+
+		restartPod = false
+		break
+	}
+
+	if !restartPod {
+		return nil
+	}
+
+	noContPodConfig := *(pod.config)
+	noContPodConfig.Containers = []ContainerConfig{}
 
-	_, err := h.hyperstart.SendCtlMessage(hyperstart.KillContainer, payload)
+	err = h.startPod(noContPodConfig)
```

Comment:
Yes sure, I will do that. I had designed it that way because I thought every container needed to be run with a pause binary, and there was no point in adding an extra pause container.